pkg/mods/file/file.go

```diff
@@ -8,16 +8,64 @@ import (
 	"src.elv.sh/pkg/eval"
 	"src.elv.sh/pkg/eval/errs"
 	"src.elv.sh/pkg/eval/vals"
+	"src.elv.sh/pkg/sys"
 )
 
 var Ns = eval.BuildNsNamed("file").
 	AddGoFns(map[string]any{
 		"close":    close,
+		"is-tty":   isTty,
 		"open":     open,
 		"pipe":     pipe,
 		"truncate": truncate,
 	}).Ns()
 
+//elvdoc:fn is-tty
+//
+// ```elvish
+// file:is-tty $file-obj?
+// ```
+//
+// Outputs `$true` if `$file-obj` is open on a tty (i.e., a terminal device);
+// otherwise, outputs `$false`. If the `$file-obj` argument is omitted the
+// default output byte stream is tested.
```

Comment:
Only supporting testing stdout is an unnecessary limitation; it can be useful to test stdin and stderr too.

`is-tty` should accept either a file object or a number representing a FD. So testing stdin, stdout and stderr can be `file:is-tty 0`, `file:is-tty 1` and `file:is-tty 2`. A FD can be converted to a os.File with (os.NewFile)[https://pkg.go.dev/os#NewFile].

This will also simplify the implementation:

- `is-tty` now only needs to deal with a single argument

- ByteOutput no longer needs to expose the underlying os.File.